src/library/dlgtrackinfo.cpp

```diff
@@ -609,20 +593,26 @@ void DlgTrackInfo::slotSpinBpmValueChanged(double value) {
     }
 
     if (!m_pBeatsClone) {
-        const mixxx::audio::FramePos cuePosition = m_pLoadedTrack->getMainCuePosition();
+        mixxx::audio::FramePos cuePosition = m_pLoadedTrack->getMainCuePosition();
+        // This should never happen, but we cannot be sure
+        VERIFY_OR_DEBUG_ASSERT(cuePosition.isValid()) {
+            cuePosition = mixxx::audio::kStartFramePos;
+        }
         m_pBeatsClone = mixxx::Beats::fromConstTempo(
                 m_pLoadedTrack->getSampleRate(),
-                cuePosition,
+                // Cue positions might be fractional, i.e. not on frame boundaries!
+                cuePosition.toNearestFrameBoundary(),
```

Comment:
Wouldn't it be more robust to handle this inside mixxx::Beats::fromConstTempo?

Reply:
My understanding was that the calling code is responsible for adjusting the actual position as needed. Implicitly rounding to frame boundaries while the calling code is unaware of these circumstances might cause unexpected inconsistencies. As long as the calling code is in control it is able to do adjustments of other values based on the rounding if needed.

Reply:
In this case we leave the original cue point as is, even if it slightly differs from the position in the beat grid.

Reply:
I don't have a strong opinion on this. @Holzhaus what do you think? Should the calling code be responsible for this rounding or should the called function handle this automatically?

Reply:
How about adding a `VERIFY_OR_DEBUG_ASSERT` inside the called function checking that the position is on a frame boundary?

Reply:
Document this assumption with a VERIFY_OR_DEBUG_ASSERT inside the called function and I'll be happy to merge this.